src/T2DMap.cpp

```diff
@@ -3040,6 +3028,10 @@ void T2DMap::mousePressEvent(QMouseEvent* event)
             connect(action, SIGNAL(triggered()), mapper, SLOT(map()));
         }
         connect(mapper, SIGNAL(mapped(QString)), this, SLOT(slot_userAction(QString)));
+
+        // After all has been added, finally have Qt display the context menu as a whole
+        mPopupMenu = true;
```

Comment:
What?

Reply:
You (well Mudlet) is creating a new `QSignalMapper` - parented to the `T2DMap` - instance each time a right click is processed and those are going to accumulate each time until they will all get `delete`d when the `T2DMap` is finally destroyed at the end of the profile's session. I am not sure that it is anything that you specifically have done but because you have tickled the code in that area I guess the analyser has just spotting it in passing.

💡 I think you ought to be able to clean this up by, instead of using a local `auto mapper` (where `auto` means the original C automatically instantiated local variable and NOT the later C++ *let the compiler work out what the hell the variable definition is*) instead declaring a `QPointer<QSignalMapper> mMapper;` in the class header file and putting the following:
```cpp
    if (mMapper) {
        mMapper->deleteLater();
        mMapper = nullptr;
    }
```
wherever there is the following line of code:
```cpp
    mPopupMenu = false;
```

HTH

Reply:
Would a local `unique_ptr` here do the job just the same but simpler?

Reply:
@Kebap this error got triggered because you edited code that had this pre-existing problem, and the clang-tidy analyser caught it. We haven't run it yet across the whole codebase to fix these kinds of issues, which is why new PRs can stumble on issues that you didn't seem to add.

Reply:
I don't think so - depends on when `mPopupMenu` is reset - if it is entirely within that method (event handler) then it might be enough but if the menu lasts longer than the end of that event handler (and I think it does) then a local `std::unique_ptr` would go out of scope at the end and delete the `QSignalMapper` instance prematurely...

Reply:
> you edited code that had this pre-existing problem

That's what I figured, too.

Out of my scope, then.

Reply:
Sounds find, mind opening an issue about this? This bit of code should be considered for refactoring with a lambda as well - it's a bit unintuitive at first look.

Reply:
You can continue your brainstorm there: #5232